### PR TITLE
Isolation calibration for EightTerm

### PR DESCRIPTION
**Pull request by Ttl**
Fixes bug #123. Implementation was actually easier than I initially thought and there are very few new lines of code. Basically isolation calibration can be done outside of the EightTerm calibration so no deep changes were needed.

Isolation calibration can also be used with other calibrations that inherit from EightTerm such as TRL, UnknownThru and NISTMultilineTRL. Isolation calibration tests were added also for these calibrations.

### CODE REVIEW

**Comment by Ttl** [comment]
Now that I took a look at the code, I don't think this method of isolation correction is completely correct. Currently in the tests isolation terms are added after the switch termination, but I think they should be added before the switch correction. Although error is probably quite small it would be good to fix it. In the "Multiline TRL Revealed" paper are listed the changes needed to be made to the switch correction to take isolation into account. I'll try to fix it later.

**Comment by Ttl** [comment]
I think that this is a correct way to implement the isolation calibration after all. It should be equivalent to the TwelveTerm isolation calibration. It does give the same results according to the "TwelveTermToEightTermTest"-test. It is not equivalent to the forward/reverse isolation terms in the SixteenTerm unless switch terms are zero, but that doesn't matter since there are also other isolation terms in the 16-term model that are not corrected with this simple isolation calibration.

This model is also given in the "Formulations of the Basic Vector Network Analyzer Error Model including Switch-Terms" paper, so I think it is okay.

I'm not sure why in the Multiline TRL paper switch terms are also modified. I don't seem to get the correct results using it. Any ideas?

**Comment by arsenovic** [comment]
thanks for working on this, i will look into mark's paper tomorrow and get back to you on my opinion.

File: skrf/calibration/calibration.py [hunk 1]
```diff
@@ -1993,12 +1998,32 @@ def __init__(self, measured, ideals, switch_terms=None,
...
+            self.isolation = measured[0].copy()
+            self.isolation.s[:,:,:] = 0
+        else:
+            self.isolation = isolation.copy()
+
+            measured = [m.copy() for m in measured]
+            for i in range(len(measured)):
+                measured[i].s[:,1,0] -= self.isolation.s[:,1,0]
+                measured[i].s[:,0,1] -= self.isolation.s[:,0,1]
+
```
Comment (arsenovic) [inline skrf/calibration/calibration.py]: this seems to correct for isolation during the init, which i think is unexpected. seems more logical that isolation is corrected for in `apply_cal`. also, i would not expect the `measured` to change. 

maybe we could make a property  `measured_unisolated` similar to `measured_terminated`. then `measured_unterminated` would `unterminate(measured_unisolated)`

**Comment by Ttl** [comment]
I think that is a better way to do it. I just added a commit to implement it.

By the way, it seems that documentation building has been broken for some time. Seems to be because of tab when spaces are expected in "readthedocs-environment.yml".

**Comment by arsenovic** [comment]
great!

**Comment by arsenovic** [comment]
i will  look into docs momentarily

**Comment by arsenovic** [comment]
it looks like the read the docs says the build fails, but they appear to be working.

**Pull request merged**